src/H5VLdyn_ops.c

```diff
@@ -67,38 +67,49 @@ static void H5VL__release_dyn_op(H5VL_dyn_op_t *dyn_op);
 /*******************/
 
 /* The current optional operation values */
-static int H5VL_opt_vals_g[H5VL_SUBCLS_TOKEN + 1] = {
-    H5VL_RESERVED_NATIVE_OPTIONAL, /* H5VL_SUBCLS_NONE */
-    H5VL_RESERVED_NATIVE_OPTIONAL, /* H5VL_SUBCLS_INFO */
-    H5VL_RESERVED_NATIVE_OPTIONAL, /* H5VL_SUBCLS_WRAP */
-    H5VL_RESERVED_NATIVE_OPTIONAL, /* H5VL_SUBCLS_ATTR */
-    H5VL_RESERVED_NATIVE_OPTIONAL, /* H5VL_SUBCLS_DATASET */
-    H5VL_RESERVED_NATIVE_OPTIONAL, /* H5VL_SUBCLS_DATATYPE */
-    H5VL_RESERVED_NATIVE_OPTIONAL, /* H5VL_SUBCLS_FILE */
-    H5VL_RESERVED_NATIVE_OPTIONAL, /* H5VL_SUBCLS_GROUP */
-    H5VL_RESERVED_NATIVE_OPTIONAL, /* H5VL_SUBCLS_LINK */
-    H5VL_RESERVED_NATIVE_OPTIONAL, /* H5VL_SUBCLS_OBJECT */
-    H5VL_RESERVED_NATIVE_OPTIONAL, /* H5VL_SUBCLS_REQUEST */
-    H5VL_RESERVED_NATIVE_OPTIONAL, /* H5VL_SUBCLS_BLOB */
-    H5VL_RESERVED_NATIVE_OPTIONAL  /* H5VL_SUBCLS_TOKEN */
+#ifdef H5_HAVE_MULTITHREAD
+/* Initialize in H5VL_init_opt_operation_table() */
+static _Atomic(int) H5VL_opt_vals_g[H5VL_SUBCLS_TOKEN + 1];
```

Comment:
To reduce the sprawl of `#ifdef H5_HAVE_MULTITHREAD`, I think it makes more sense to create an `H5_ATOMIC` macro that is conditionally defined to `_Atomic` when `H5_HAVE_MULTITHREAD` is defined and nothing when it isn't.